source/common/ssl/context_config_impl.h

```diff
@@ -65,13 +71,13 @@ class ContextConfigImpl : public virtual Ssl::ContextConfig {
   const std::string cipher_suites_;
   const std::string ecdh_curves_;
 
-  Ssl::TlsCertificateConfigPtr tls_certificate_config_;
+  std::vector<Ssl::TlsCertificateConfigImpl> tls_certificate_configs_;
```

Comment:
Aren't we losing `std::unique_ptr` wrapping here?

Reply:
Yeah, but `std::vector` is RAII-aware, see https://en.cppreference.com/w/cpp/language/raii.

Reply:
Hmm... I think we're using `std::vector<TypePtr>` in majority of the codebase...

But if you say it's fine, then it's fine :) Thanks!

Reply:
Yeah, in general you can only do this if the class has a sensible copy/move constructor to allow vector to grow. In this case, `TlsCertificateConfigImpl` is just a bunch of const strings, so it works fine.